src/Apple/Provider.php

```diff
@@ -88,12 +106,53 @@ public function getAccessTokenResponse($code)
      */
     protected function getUserByToken($token)
     {
-        static::verify($token);
+        $this->checkToken($token);
         $claims = explode('.', $token)[1];
 
         return json_decode(base64_decode($claims), true);
     }
 
+    protected function getClientSecret()
+    {
+        if (!$this->jwtConfig) {
+            $this->getJwtConfig(); // Generate Client Secret from private key if not set.
+        }
+
+        return $this->clientSecret;
+    }
+
+    protected function getJwtConfig()
+    {
+        if (!$this->jwtConfig) {
+            $private_key_path = $this->getConfig('private_key', '');
+            $private_key_passphrase = $this->getConfig('passphrase', '');
+            $signer = $this->getConfig('signer', '');
+
+            if (empty($signer) || !class_exists($signer)) {
+                $signer = !empty($private_key_path) ? \Lcobucci\JWT\Signer\Ecdsa\Sha256::class : AppleSignerNone::class;
+            }
+
+            if (!empty($private_key_path) && file_exists($private_key_path)) {
+                $this->privateKey = file_get_contents($private_key_path);
+            } else {
+                $this->privateKey = $private_key_path; // Support for plain text private keys
+            }
+
+            $this->jwtConfig = Configuration::forSymmetricSigner(
+                new $signer(),
+                AppleSignerInMemory::plainText($this->privateKey, $private_key_passphrase)
+            );
+
+            if (!empty($this->privateKey)) {
+                $appleToken = new AppleToken($this->getJwtConfig());
```

Comment:
Not too happy about this recursive call. The config is already created a few lines earlier, and it does seem like it would work if used directly.